File: stack_queue/12906.py
# 같은 숫자는 싫어
def solution(arr):
    answer = [arr[0]]
    
    idx_answer = 0
    for i in range(1, len(arr)):
        if arr[i] == answer[idx_answer]: continue
        else: answer.append(arr[i]); idx_answer += 1
    
    return answer

File: stack_queue/12906.cpp
// 같은 숫자는 싫어

#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;

    int first_check = 1;
    for(auto iter=arr.begin(); iter!=arr.end(); iter++) {
        if(first_check == 1) { answer.push_back(*iter); first_check = 0; continue; }
        
        if(answer.back() == *iter) continue;
        else answer.push_back(*iter);
    }

    return answer;
}